### ISSUE TITLE
Segmentation fault when using too many processors for parallel calculation

### ISSUE THREAD

**Issue by BJEANNOT0** [usgs/pestpp#194]
Hello,

I run pestpp-ies on a HPC cluster. I noticed that when I use too many processors for calculation in parallel, I get the following error :
manager.sh: line 4: 2831431 Segmentation fault      (core dumped) ./pestpp-ies case /h :27318

Also, a .core file is produced to state what the segmentaiton fault is. I am not an expert in reading such files but I managed to at least partly read it using the gdb program. I got the following message : 
#0 0x000000000046a6f4 in RunManagerPanther::get_agent_stats() ()

I also send as an attachment my case.rmr file, my case.pst file and the panther rec file of the last agent who interacted with the manager. It seems the error happens when the agent responds to REQ_RUNDIR
[panther_worker.txt](https://github.com/usgs/pestpp/files/8919236/panther_worker.txt)
[case.rmr.txt](https://github.com/usgs/pestpp/files/8919250/case.rmr.txt)
[case.pst.txt](https://github.com/usgs/pestpp/files/8919256/case.pst.txt)

I emphasize that my manager has a LOT of memory available and that every agent has the same memory available. As a consequence I do not really understand where the error could come from. Do you have any idea about this ?
Thank you for your help.

Best regards,

**Comment by jtwhite79** [comment]
Looks like you are using an older version.  Can you try this same run with the latest 5.1.17?

**Comment by jtwhite79** [comment]
I see something else.  We've found that on linux, unless you can get the system level max connections and max file handles, etc, (you can google "ulimit" and "max file handles" to get a sense of this) increased, the OS doesnt like more than about 1024 workers - its something OS related.  Ive done 2.5K+ workers on windows without an issue, but every linux flavor Ive tried has the 1024 limit...

**Comment by BJEANNOT0** [comment]
Thank you for your very pertinent insights, delevered in no time after i asked my question.
I noticed indeed that 1024 was around the limit - and I am using Linux. I will check If I can modify the system max connections and handles on the HPC I am using.
Also, I did not realize that I was stuck with such an old version of the software ==> I update right now.

I think we can close the issue. I will reopen if I end up succeeding going over 1024 to share the way I did it.
Thank you again.
